### ISSUE TITLE
Invariant Violation: requireNativeComponent: "RNSVGSvgViewAndroid" was not found in the UIManager in Android

### ISSUE THREAD

**Issue by Nikita-Chernenko** [software-mansion/react-native-svg#1944]
# Bug

When I use "react-native-svg": "^13.6.0", I get the following error on Android
Invariant Violation: requireNativeComponent: "RNSVGSvgViewAndroid" was not found in the UIManager in Android
When I downgrade to 13.4.0, it works fine

**Comment by carlaTatiana** [comment]
I have the same issue. 

If I install react-native-svg 13.6.0, I have this issue running the app in Expo Go: "Invariant Violation: requireNativeComponent: "RNSVGSvgViewAndroid" was not found in the UIManager in Android"
But works fine if a create a build.

If I install react-native-svg 13.4.0 I have this issue when I run a build (Android)
 
![Captura de Pantalla 2022-12-13 a la(s) 12 52 44](https://user-images.githubusercontent.com/72270207/208450204-4a21faac-7f02-4a03-ac4b-6db451427a5c.png)

But if I run the app on Expo Go, everything is ok

**Comment by rajsawhoney** [comment]
And the interesting thing is it works with no issue on dev-client build as well i.e with` "developmentClient": true`.
When I build with `"developmentClient": false` the app crashes to the screen where I am dealing with react-native-svg stuffs.

**Comment by heathercodes** [comment]
I was also encountering this issue, but I think its related to [this update](https://github.com/software-mansion/react-native-svg/pull/1847) from the 13.6.0 release. If you're still on a React Native version lower than 0.68/0.69 or so, you'll likely encounter this because you're not using Fabric and I don't think the update made in that PR is backwards compatible. Also note that the PR links to this [PR](https://github.com/software-mansion/react-native-svg/pull/1913/files), that updates the support table to indicate that 13.6.0 requires React Native 0.70.

Downgrading to 13.4.0 fixed the issue for me. I'm using Yarn, so I had to make sure the yarn.lock package was now resolving to 13.4.0 instead of 13.6.0.

**Comment by WoLewicki** [comment]
I posted the explanation on [SO thread](https://stackoverflow.com/questions/74756210/invariant-violation-requirenativecomponent-rnsvgsvgviewandroid-was-not-found), but I will copy paste it here:
You cannot use "react-native-svg": "13.6.0" with expo sdk 47 Managed Workflow because the native code is compiled there already, and it is compiled with "react-native-svg": "13.4.0" version of native code. Between those versions, the naming of native component changed, so when using "react-native-svg": "13.6.0", you update only the JS code, which tries to get new name of component from the native side, but the native code still has the previous name compiled. It resolves in such an error. Unfortunately there is no solution to this problem in managed workflow, the easiest solution would be to use EAS (https://expo.dev/eas) to build custom version of Expo Go app with "react-native-svg": "13.6.0" compiled in it.

@heathercodes as for the support, we 13.6.0 requires React Native 0.70 when building with the new architecture, on old arch it should work correctly on older version too (see https://github.com/software-mansion/react-native-svg#supported-react-native-versions).

Does it solve your issue?

**Comment by robimuller** [comment]
I have the same issue. `requireNativeComponent:RNSVGsvgViewAndroid was not found in the UIManager`

@WoLewicki  I am using RN 0.70 and the rnsvg seems to be 13.6.0, so to answer your question, it unfortunately doesnt work regardless the version.  I have no clue whats wrong...

**Comment by robimuller** [comment]
> I have the same issue. `requireNativeComponent:RNSVGsvgViewAndroid was not found in the UIManager`
> 
> @WoLewicki I am using RN 0.70 and the rnsvg seems to be 13.6.0, so to answer your question, it unfortunately doesnt work regardless the version. I have no clue whats wrong...



> I posted the explanation on [SO thread](https://stackoverflow.com/questions/74756210/invariant-violation-requirenativecomponent-rnsvgsvgviewandroid-was-not-found), but I will copy paste it here: You cannot use "react-native-svg": "13.6.0" with expo sdk 47 Managed Workflow because the native code is compiled there already, and it is compiled with "react-native-svg": "13.4.0" version of native code. Between those versions, the naming of native component changed, so when using "react-native-svg": "13.6.0", you update only the JS code, which tries to get new name of component from the native side, but the native code still has the previous name compiled. It resolves in such an error. Unfortunately there is no solution to this problem in managed workflow, the easiest solution would be to use EAS (https://expo.dev/eas) to build custom version of Expo Go app with "react-native-svg": "13.6.0" compiled in it.
> 
> @heathercodes as for the support, we 13.6.0 requires React Native 0.70 when building with the new architecture, on old arch it should work correctly on older version too (see https://github.com/software-mansion/react-native-svg#supported-react-native-versions).
> 
> Does it solve your issue?

I have the same issue. requireNativeComponent:RNSVGsvgViewAndroid was not found in the UIManager

@WoLewicki I am using RN 0.70 and the rnsvg seems to be 13.6.0, so to answer your question, it unfortunately doesnt work regardless the version. I have no clue whats wrong...

**Comment by WoLewicki** [comment]
Do you use Expo Go @robimuller ?

**Comment by robimuller** [comment]
> Do you use Expo Go @robimuller ?

yes

**Comment by robimuller** [comment]
> Do you use Expo Go @robimuller ?

yes @WoLewicki , do you have any suggestions? :/

**Comment by WoLewicki** [comment]
As mentioned here:
> Unfortunately there is no solution to this problem in managed workflow, the easiest solution would be to use EAS ([expo.dev/eas](https://expo.dev/eas)) to build custom version of Expo Go app with "react-native-svg": "13.6.0" compiled in it.

it seems like the best solution for now.

**Comment by cprentice22** [comment]
Has there been a fix to this problem? I am still having the same issue as the title. On "react-native": "0.70.5" and "react-native-svg": "13.4.0". I have also tried 13.6 but it still does not work. When I test on an android phone the bundling completes and right after the splash screen loads the app crashes every time. @heathercodes @robimuller , have you found a fix for this problem?

**Comment by WoLewicki** [comment]
@cprentice22 are you using Expo Go app?

**Comment by manhhoang02** [comment]
> @cprentice22 are you using Expo Go app?

do you have any suggestions?

**Comment by cprentice22** [comment]
> > @cprentice22 are you using Expo Go app?
> 
> do you have any suggestions? @WoLewicki

I noticed in my package.json it said "react-native-svg": "^13.4.0". Because of the up facing arrow in front of the version number it was updating to the latest version all of the time. I took out the arrow so in my package.json it says "react-native-svg": "13.4.0" and this fixed my problem for now. As far as I can tell any later versions still don't work with android.

**Comment by meninjanho4788** [comment]
@cprentice22 i downgrade to react-native-svg 13.4.0, but it have another error
"Tried to register two views with the same name RNSVGRect"

**Comment by cprentice22** [comment]
> 

I did what I said in my comment above and then npm uninstall react-native-svg. Then npm install react-native-svg@13.4.0. Make you do both things. This comment and the last and kill and restart your app and it will work.

**Comment by liptonzuma** [comment]
This issue is  still around with latest sdk of expo. ios works fine but android.

**Comment by WoLewicki** [comment]
For some reason, the version of `react-native-svg` has not been bumped in the newest SDK version, so you should stick with `13.4.0` when using Expo Go. I'll close this issue since it is not something we can fix in the library's code, but rather a matter of Expo team upgrading the package, if you want to use the newest version of the library. Feel free to comment here if something is wrong and we can always reopen it then.

**Comment by da45** [comment]
just run "npx expo install react-native-svg react-native-qrcode-svg" instead of "npm i -S" as  mentionned in official doc

**Comment by jwallet** [comment]
> just run "npx expo install react-native-svg react-native-qrcode-svg" instead of "npm i -S" it works fine for me

yes in a managed workflow adding the package through npx expo actually works

```
npx expo install react-native-svg
yarn expo start --clear
```

**Comment by LushawnDev** [comment]
Just adding to this thread with my experience for people seeing this now.

I recently upgraded expo to `49.0.21`. I had `"react-native-svg": "13.6.0"` installed, I upgraded to `13.9.0` which fixed my issue.

An obvious "clue" but expo was giving me warnings to upgrade a few packages which were not incompatible with my expo version, so check for those.
